File: spring-boot-database-cache-example/src/main/java/com/javadeveloperzone/service/EmployeeService.java
package com.javadeveloperzone.service;

import com.javadeveloperzone.model.Employee;

import java.util.List;

/**
 * Created by JavaDeveloperZone on 04-04-2018.
 */

public interface EmployeeService {

    List<Employee> findAll();
    Employee save(Employee employee);
    void delete(long employeeId);
}

File: spring boot example + angular/spring-boot-ui/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Subscribable, Observable } from 'rxjs';
import { Employee } from '../model/employee';
@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  constructor(private http: HttpClient) { 
    
  }
  public save(employee:Employee):Observable<any>{
    return this.http.post('http://localhost:8080/save',employee);
  }
  public update(employee:Employee):Observable<any>{
    return this.http.post('http://localhost:8080/update',employee);
  }
  public getEmployeeList():Observable<any>{
    return this.http.get('http://localhost:8080/list');
  }
  public deleteEmployee(employeeId:number):Observable<any>{
    return this.http.delete(`http://localhost:8080/delete?id=${employeeId}`)
  }
  public findById(employeeId:number):Observable<any>{
    return this.http.get(`http://localhost:8080/findById?id=${employeeId}`)
  }
}
